### PR TITLE
LogFactory.Setup - SetupInternalLogger

### PR DESCRIPTION
**Pull request by snakefoot**
Created as follow up to #3686

### CODE REVIEW

File: src/NLog/SetupInternalLoggerBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    /// Extension methods to setup NLog <see cref="InternalLogger"/> options
+    /// </summary>
+    public static class SetupInternalLoggerBuilderExtensions
+    {
+        /// <summary>
+        /// Configures <see cref="InternalLogger.LogLevel"/>
+        /// </summary>
+        public static ISetupInternalLoggerBuilder EnableLogLevel(this ISetupInternalLoggerBuilder setupBuilder, LogLevel logLevel)
+        {
+            InternalLogger.LogLevel = logLevel;
```
Comment (304NotModified) [inline src/NLog/SetupInternalLoggerBuilderExtensions.cs]: Enable here means enable for **minimal** loglevel, isn't? Then maybe the name is confusing compared to the logging rule. 

(PS only mobile now, so hard to check code)

Proposal EnableForMinimumLogLevel.

File: tests/NLog.UnitTests/Config/LogFactorySetupTests.cs [hunk 2]
```diff
@@ -284,5 +285,114 @@ public void SetupExtensionsRegisterLayoutRendererTypeTest()
...
+                Assert.Equal(logLevel, InternalLogger.LogLevel);
+            }
+            finally
+            {
+                InternalLogger.Reset();
+            }
+        }
+
+        [Fact]
+        public void SetupInternalLoggeLogToFileTest()
```
Comment (304NotModified) [inline tests/NLog.UnitTests/Config/LogFactorySetupTests.cs]: Logge ;)

File: src/NLog/SetupInternalLoggerBuilderExtensions.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/SetupInternalLoggerBuilderExtensions.cs]: Changed to `SetMinimumLogLevel`

File: tests/NLog.UnitTests/Config/LogFactorySetupTests.cs (hunk 2 above)
Comment (snakefoot) [reply tests/NLog.UnitTests/Config/LogFactorySetupTests.cs]: Fixed

File: src/NLog/SetupInternalLoggerBuilderExtensions.cs (hunk 1 above)
Comment (304NotModified) [reply src/NLog/SetupInternalLoggerBuilderExtensions.cs]: Even better :+1:

**Comment by 304NotModified** [comment]
There seems to be a build error

**Comment by snakefoot** [comment]
> There seems to be a build error

Seems you are right. Trying again :)

**Comment by 304NotModified** [comment]
:+1:

**Comment by snakefoot** [comment]
Updated wiki: https://github.com/NLog/NLog/wiki/Configure-from-code#logfactory-initial-setup

**Pull request merged**